### DISCUSSION TITLE
Error When Deploying to Sandbox

### DISCUSSION THREAD

**Discussion by nicward** [microsoft/AL-Go#534] [Q&A] [answered]
Hello,
I'm working through these steps: https://freddydk.github.io/AL-Go/

I seem to be stuck on step 3, I performed all the prerequisites but get this error when trying to run the workflow 'Publish To Environment'

**Deploy action failed. Error: The property 'tenantId' cannot be found on this object.**

I cut and paste this to a Secret called 'AUTHCONTEXT' and added the required information.

{"TenantID":"<TenantID>","ClientID":"<theClientID>","ClientSecret":"<theClientSecret>"}

I can't see what I have done wrong, but I'm sure I have.

I would appreciate any help or advice please.

Thanks

Nick

**Comment by freddydk** [comment]
probably need the full output of the job to see what might be wrong.

**Reply by nicward** [reply]
Ok, is this what is needed:

Run microsoft/AL-Go-Actions/Deploy@v3.0
Run try { D:\a\_actions\microsoft\AL-Go-Actions\v3.0\Deploy/Deploy.ps1 -actor $ENV:_actor -token $ENV:_token -parentTelemetryScopeJson $ENV:_parentTelemetryScopeJson -projects $ENV:_projects -environmentName $ENV:_environmentName -artifacts $ENV:_artifacts -type $ENV:_type } catch { Write-Host "::Error::Unexpected error when running action ($($_.Exception.Message.Replace("`r",'').Replace("`n",' ')))"; exit 1 }
Downloading BcContainerHelper latest version from CDN
BcContainerHelper version 5.0.1
BC.HelperFunctions emits usage statistics telemetry to Microsoft
Analyzing artifacts
https://api.github.com/repos/HexMI/app1/actions/artifacts?per_page=100&page=1
project '*'
Analyzing artifacts
https://api.github.com/repos/HexMI/app1/actions/artifacts?per_page=100&page=1
project '*'
Downloading artifact app1-main-Apps-1.0.3.0
https://api.github.com/repos/HexMI/app1/actions/artifacts/694298151/zip
Apps to deploy
D:\a\app1\app1\.artifacts\app1-main-Apps-1.0.3.0.zip
Attempting authentication to https://api.businesscentral.dynamics.com/.default using clientCredentials...
The remote server returned an error: (401) Unauthorized. invalid_client
AADSTS7000215: Invalid client secret provided. Ensure the secret being sent in the request is the client secret value, not the client secret ID, for a secret added to app '***'.
Trace ID: 96693891-a96a-4469-93e1-e71399d20200
Correlation ID: df7d151a-2298-4[195](https://github.com/HexMI/app1/actions/runs/4966235753/jobs/8888133382#step:7:197)-81b6-819e9a68566a
Timestamp: [202](https://github.com/HexMI/app1/actions/runs/4966235753/jobs/8888133382#step:7:204)3-05-13 11:17:41Z
Unauthorized


Authentication failed
Error: Deploy action failed. Error: The property 'tenantId' cannot be found on this object. Verify that the property exists. Stacktrace: at <ScriptBlock>, D:\a\_actions\microsoft\AL-Go-Actions\v3.0\Deploy\Deploy.ps1: line 129 at <ScriptBlock>, D:\a\_temp\42bb4862-bd65-43ec-af8d-720213776e61.ps1: line 2 at <ScriptBlock>, <No file>: line 1
AL-Go action ran: Deploy Telemetry Correlation Id: 9804c480-c4f7-4ec3-a932-2373d9c87d0e
Removing BcContainerHelper
Error: Process completed with exit code 1.

**Comment by freddydk** [comment, marked as answer]
Yeah, this is the error coming from AAD:
```
Attempting authentication to https://api.businesscentral.dynamics.com/.default using clientCredentials...
The remote server returned an error: (401) Unauthorized. invalid_client
AADSTS7000215: Invalid client secret provided. Ensure the secret being sent in the request is the client secret value, not the client secret ID, for a secret added to app '***'.
```
Will see if I can change the AL-Go error to include this - looks like a follow-up error.

So, your secret is wrong, probably correctly formatted, but the values are wrong.

**Reply by nicward** [reply]
Thank you!

Apologies, for such a basic mistake, hopefully I will get the hang of this soon.

Nick

**Reply by freddydk** [reply]
No need for apology - will take a note that the docs can be better:-)
